docs/AUTOMATED_TESTS.md

```diff
@@ -0,0 +1,37 @@
+# Automated Tests
+
+DAPLink has an extensive automated test suite. It can be run on the current development environment or a release package. Test results are written to the current directory and printed to the console.
+
+Tests are launched by running ``python test\run_test.py`` in the Python virtual environment. The test logic discovers connected DAPLink boards and targets only those boards. Likewise, it detects what firmware you have built and targets only that firmware.
+
+The tests rely on information in ``test/info.py``. If you are porting DAPLink to a new board, you will need to add to info.py. See [Adding a new board](PORT_BOARD.md) for details.
+
+### Two Areas of Testing
+Tests are grouped into two categories
+
+* Tests that validate how DAPLink is used for programming the target device and even the HIC itself via MSD--by copying an image file to the MSD drive (most users will drag-and-drop). These tests do not run by default but can be turned on by specifying --testdl.
+
+
+* Tests that validate the MSD, CDC, and CMSIS-DAP endpoints. These tests run by default, but can be turned off by specifying --notestendpt
+
+The endpoint tests rely on a target application--a small UART program that implements a simple protocol. The target app is used by all three endpoint sub-tests.
+
+For mbed boards that are registered on mbed.org, the test builds the target application code on the fly in the cloud using the mbed RESTful Compile API. It then downloads the resulting image and downloads it to the target. The UART application source is in an mbed git repository (https://developer.mbed.org/users/c1728p9/code/daplink-validation/). For this workflow, you must specify your mbed.org user id and password via --user and --password.
```

Comment:
Currently, the project is in a mercurial repository on mbed, not git.  What do you think about something like this:
```
The UART application source is the mbed mercurial repository [daplink-validation](https://developer.mbed.org/users/c1728p9/code/daplink-validation/)
```
Without the code formatting:
The UART application source is the mbed mercurial repository [daplink-validation](https://developer.mbed.org/users/c1728p9/code/daplink-validation/)